rustls/src/msgs/deframer/handshake.rs

```diff
@@ -0,0 +1,452 @@
+use alloc::vec::Vec;
+use core::mem;
+use core::ops::Range;
+
+use super::buffers::{Coalescer, Delocator, Locator};
+use crate::msgs::codec::{u24, Codec};
+use crate::msgs::message::InboundPlainMessage;
+use crate::{ContentType, ProtocolVersion};
+
+#[derive(Debug)]
+pub(crate) struct HandshakeDeframer {
+    /// Spans covering individual handshake payloads, in order of receipt.
+    spans: Vec<FragmentSpan>,
+}
+
+impl HandshakeDeframer {
+    /// Accepts a message into the deframer.
+    ///
+    /// `containing_buffer` allows mapping the message payload to its position
+    /// in the input buffer, and thereby avoid retaining a borrow on the input
+    /// buffer.
```

Comment:
Perhaps helpful to lift some of the later context w.r.t _why_ retaining a borrow is problematic into this part of the comment?